src/nvim/lua/vim.lua

```diff
@@ -212,6 +212,19 @@ do
         vim.api.nvim_put(lines, 'c', true, true)
         -- XXX: Normal-mode: workaround bad cursor-placement after first chunk.
         vim.api.nvim_command('normal! a')
+      elseif phase < 2 and mode == 'R' then
+        local nchars = 0
+        for _, line in ipairs(lines) do
+            nchars = nchars + line:len()
+        end
+        local pos = vim.api.nvim_win_get_cursor(0)
+        local row, col = pos[1], pos[2]
+        local line = vim.api.nvim_buf_get_lines(0, row-1, row, true)[1]
```

Comment:
This is the best we can do for now but we should use `nvim_buf_set_text` #11833 later when it is available

Reply:
I agree. I think there are lots of places where that API can simplify code, so some grepping and refactoring when it is ready would probably be a good idea.